api/v1alpha1/clusterextension_types.go

```diff
@@ -158,3 +159,21 @@ type ClusterExtensionList struct {
 func init() {
 	SchemeBuilder.Register(&ClusterExtension{}, &ClusterExtensionList{})
 }
+
+func (r *ClusterExtension) GetPackageSpec() *ExtensionSourcePackage {
+	p := &ExtensionSourcePackage{}
+
+	p.Channel = r.Spec.Channel
+	p.Name = r.Spec.PackageName
+	p.Version = r.Spec.Version
+
+	return p
+}
+
+func (r *ClusterExtension) GetUID() types.UID {
+	return r.ObjectMeta.GetUID()
+}
+
+func (r *ClusterExtension) GetUpgradeConstraintPolicy() UpgradeConstraintPolicy {
+	return r.Spec.UpgradeConstraintPolicy
+}
```

Comment:
Drop this since UpgradeConstraintPolicy is now part of the package struct?

Reply:
The interface function is not being used, so I'm removing it